### DISCUSSION TITLE
How to delay the animation start until some state value has been initialized?

### DISCUSSION THREAD

**Discussion by shenghan97** [nandorojo/moti#326] [Q&A]
I have a component that looks like this:
```JSX
<MotiView     
  animate={{
  width: mainInputContentSize }} >
    <TextInput
       onContentSizeChange={(event) => {
           setMainInputContentSize( event.nativeEvent.contentSize.width )}}
       placeholder = { i18n.placeholder }
    />             
</MotiView>
```

I need the MotiView to animate its' width dynamically as the users type. However. because of i18n, the initial width of the textinput is unknown ahead of time. This results in an unnecessary animation when the component first loads, from a default state value to the correct i18n value.

I've tried to use a tertiary expression to not set the width in the animate props until the state got initialized, but that doesn't seem to change anything. 

Is there any way I can delay the start of the animation until the state value has been set? 

Any idea is welcome! Thanks!

**Comment by nandorojo** [comment]
You could use transition.type = ‘no-animation’ conditionally

**Reply by shenghan97** [reply]
> You could use transition.type = ‘no-animation’ conditionally 

Thanks for the reply! But I also tried with no-animation and it's doesn't work either. 

But i ended up finding out that it will work if I use a setTimeout to delay a no-animation condition with 1ms delay within a useEffect. 

I'm still not sure why simply using useEffect to set no-animation without timeout won't work. Got any idea?
